### PR TITLE
feat: Add support for running command with arguments in containers

### PR DESCRIPTION
**Pull request by nils1k**
I was a bit unsure with respect to where in values.yaml it would make sense to add command and args. Feel free to come with any corrections you deem necessary.

### CODE REVIEW

File: charts/common/templates/cron.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ spec:
         - name: {{ .name | default $app }}
           image: {{ $image }}
           imagePullPolicy: Always
+          command: {{ .Values.common.container.command }}
+          args: {{ .Values.common.container.args }}
```
Comment (kagkarlsson) [inline charts/common/templates/cron.yaml:47]: We need to add it for the deployment as well. 

Additionally, I think it should only be added if it is set in the values file

File: charts/common/values.yaml [hunk 2]
```diff
@@ -179,6 +179,10 @@ container:
   volumes: []
   # -- (int) Override pod terminationGracePeriodSeconds (default 30s).
   terminationGracePeriodSeconds:
+  # -- Add command that will run in the pod. Accepts Kubernetes syntax.
+  command:
+  # -- Add arguments that will be passed to the command. Accepts Kubernetes syntax.
+  args:
```
Comment (kagkarlsson) [inline charts/common/values.yaml:186]: I suggest adding it under

```
  # -- Name of container
  # @default -- .app
  name:
  # -- Add labels to your pods
  labels: {}
  <HERE>
  ```

File: charts/common/templates/cron.yaml (hunk 1 above)
Comment (nils1k) [reply charts/common/templates/cron.yaml:47]: Thanks. So, something like this? -->

```suggestion
          {{- if .Values.common.container.command -}}
          command: {{ .Values.common.container.command }}
           {{- end }}
           {{- if .Values.common.container.args -}}
           args: {{ .Values.common.container.args }}
           {{- end }}
```

Comment (kagkarlsson) [reply charts/common/templates/cron.yaml:47]: I think it's
`.Values.common.container.args` -> `.args`  (and similar for command)

(and indentation slightly off ;) )

**Comment by nils1k** [comment]
> We should also add a tests (test that no command will not list, and that a command will render as intended) both for cron and deployment?
> 
> Or do we scope for just the cron in this PR?

Good point. I think we can add support for both resource types , and I'll add tests for the changes.

**Comment by kagkarlsson** [comment]
> Or do we scope for just the cron in this PR?

I think it needs to be supported for deployments aswell

File: charts/common/templates/cron.yaml [hunk 3]
```diff
@@ -43,6 +43,12 @@ spec:
         - name: {{ .name | default $app }}
           image: {{ $image }}
           imagePullPolicy: Always
+          {{- if .command -}}
+            command: {{ .Values.common.container.command }}
+           {{- end }}
+           {{- if .args -}}
+            args: {{ .Values.common.container.args }}
+           {{- end }}
```
Comment (kagkarlsson) [inline charts/common/templates/cron.yaml:51]: I think it should be like this (might be wrong though):

```          
imagePullPolicy: Always
{{- if .command -}}
command: {{ .command }}
{{- end }}
{{- if .args -}}
args: {{ .args }}
{{- end }}
```

File: charts/common/tests/cron_test.yaml [hunk 4]
```diff
@@ -27,7 +27,7 @@ values: &values
       version: 1
   containers:
     - name: test
-      image: img
+      image: img
```
Comment (AlexanderBrevig) [inline charts/common/tests/cron_test.yaml:30]: remove trailing whitespace?

File: charts/common/values.yaml [hunk 5]
```diff
@@ -101,6 +101,10 @@ container:
   name:
   # -- Add labels to your pods
   labels: {}
+  # -- Add command that will run in the pod. Accepts Kubernetes syntax.
```
Comment (AlexanderBrevig) [inline charts/common/values.yaml:104]: Change `Add` for `Set the` for both `command` and `args`.

**Review by AlexanderBrevig — DISMISSED** [review]

File: charts/common/tests/cron_test.yaml (hunk 4 above)
Comment (AlexanderBrevig) [reply charts/common/tests/cron_test.yaml:30]: nitpick

File: charts/common/values.yaml (hunk 5 above)
Comment (AlexanderBrevig) [reply charts/common/values.yaml:104]: nitpick

**Review by kagkarlsson — APPROVED** [review]

**Pull request merged**